var/spack/repos/builtin/packages/ispc/package.py

```diff
@@ -60,6 +61,11 @@ class Ispc(CMakePackage):
         sha256="d3ccf547d3ba59779fd375e10417a436318f2200d160febb9f830a26f0daefdc",
     )
 
+    def setup_build_environment(self, env):
+        if self.spec.version >= Version("1.18.0"):
```

Comment:
I agree, in general it is a good habit to get into when checking the package's spec to use `self.spec.satisfies()`.

Reply:
Thank you for putting in the work to build this!

In the meantime, LLVM 16 seems to have hit develop, but this breaks the build for ispc, as it is only compatible with LLVM 13-15. Because of this, I would like to add at least https://github.com/aumuell/spack/commit/79432e2830de49cce6f70fc0bb001720d9f9cefd to this PR. Unfortunately, I do not know how to do that. (I tried pushing to your ispc branch and also could not find how to make a PR to your ispc branch).

If you are ok with including my additional changes from https://github.com/aumuell/spack/tree/heerener-ispc, please pull them into your branch. But I think that this should get merged either way.

Reply:
Thanks for the clarifications! I've managed to create the PR by starting from your fork.
Where did you find what versions of llvm ispc requires? Because the version restrictions I set were mostly guesswork.

Reply:
> Where did you find what versions of llvm ispc requires? Because the version restrictions I set were mostly guesswork.

Unfortunately I don't know where this information can be looked up. That 1.19 is only compatible with LLVM 13-15 was printed to me while it was trying to build against LLVM 16. For other versions of ispc I don't know either.

Reply:
Ah! I've seen that same message for one of the versions, just didn't think to make it fail on purpose for the others :-)